#!/usr/bin/env python

PACKAGE = "mujoco_contact_surface_sensors"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("update_rate", double_t, 0, "Update rate of the sensor in Hz", 100.0)
gen.add("visualize", bool_t, 0, "Visualize the sensor", False)
gen.add("use_parallel", bool_t, 0, "Use OpenMP parallelization", False)
gen.add("resolution", double_t, 0, "How far apart centers of the sensor cells are from each other", 0.01)
gen.add("sampling_resolution", int_t, 0, "How many samples to take per sensor cell", 20)

window_enum = gen.enum([
    gen.const("NONE", int_t, 0, "No windowing"),
    gen.const("GAUSS", int_t, 1, "Gaussian windowing"),
    gen.const("TUKEY", int_t, 2, "Tukey windowing"),
    gen.const("SQUARE", int_t, 3, "Squared distance windowing")
], "Enum for windowing functions")

gen.add("window", int_t, 0, "Windowing function to use", 0, 0, 3, edit_method=window_enum)
gen.add("sigma", double_t, 0, "Sigma for Gaussian or Tukey windowing", 0.1)

exit(gen.generate(PACKAGE, "tactile_surface_sensor", "DynamicFlatTactile"))
